graphql_compiler/compiler/emit_sql.py

```diff
@@ -74,31 +77,244 @@ def _find_columns_used(sql_schema_info, ir):
+        """Combine all parts of the fold object to produce the complete fold subquery."""
+        return sqlalchemy.select(
+            self.outputs
+        ).select_from(
+            subquery_from_clause
+        ).group_by(
+            *self.group_by
+        )
+
+    def _get_fold_outputs(self, fold_scope_location, join_descriptor, all_folded_outputs):
+        """Generate output columns for innermost fold scope and add them to active SQLFoldObject."""
+        # find outputs for this fold in all_folded_outputs and add to self._outputs
+        if fold_scope_location.fold_path in all_folded_outputs:
+            for fold_output in all_folded_outputs[fold_scope_location.fold_path]:
+                # distinguish between folds with the same fold path but different query paths
+                if (fold_output.base_location, fold_output.fold_path) == (
+                        fold_scope_location.base_location, fold_scope_location.fold_path):
+                    if fold_output.field == '_x_count':
+                        raise NotImplementedError(u'_x_count not implemented in SQL')
+
+                    # explicit label forces column to have label as opposed to anon_label
+                    intermediate_fold_output_name = 'fold_output_' + fold_output.field
+                    # add array aggregated output column to self._outputs
+                    self._outputs.append(
+                        sqlalchemy.func.array_agg(
+                            self.output_vertex_alias.c[fold_output.field]
+                        ).label(intermediate_fold_output_name)
+                    )
+
+        # this is the unique identifier for the outer vertex used to join to the outer table
+        self._outputs.append(self.outer_vertex_alias.c[join_descriptor.from_column])
+
+        return sorted(self._outputs, key=lambda column: column.name, reverse=True)
+
+    def visit_output_vertex(self,
+                            output_alias,
+                            fold_scope_location,
+                            join_descriptor,
+                            all_folded_outputs):
+        """Update output and group by columns when visiting the vertex containing output columns."""
+        self._output_vertex_alias = output_alias
+        self._outputs = self._get_fold_outputs(fold_scope_location,
+                                               join_descriptor,
+                                               all_folded_outputs)
+
+    def visit_traversed_vertex(self, join_descriptor, from_table, to_table):
+        """Add a new join descriptor for every vertex traversed in the fold."""
+        self._join_info.append((join_descriptor, from_table, to_table))
+
+    def end_fold(self, alias_generator, from_clause, outer_from_table):
+        """Produce the final subquery and join it onto the rest of the query."""
+        # for now we only handle folds containing one traversal (i.e. join)
+        edge, from_alias, to_alias = self.join_info.pop()
```

Comment:
Should we raise a `NotImplementedError` if `self.join_info` has more than 1 join?